src/app/feed/feed-advanced-search/feed-advanced-search.component.spec.ts

```diff
@@ -30,7 +30,156 @@ describe('Component: FeedAdvancedSearchComponent', () => {
+		const fixture = TestBed.createComponent(FeedAdvancedSearchComponent);
+		const component = fixture.debugElement.componentInstance;
+		component.getTimeBoundResults('any');
+
+		expect(component.timeBoundValue).toBe('Any time');
+	}));
+
+	it('should call getTimeBoundResults(lastDay)', async(() => {
+		const fixture = TestBed.createComponent(FeedAdvancedSearchComponent);
+		const component = fixture.debugElement.componentInstance;
+		component.getTimeBoundResults('lastDay');
+
+		expect(component.timeBoundValue).toBe('Past 24 hours');
+	}));
+
+	it('should call getTimeBoundResults(lastWeek)', async(() => {
+		const fixture = TestBed.createComponent(FeedAdvancedSearchComponent);
+		const component = fixture.debugElement.componentInstance;
+		component.getTimeBoundResults('lastWeek');
+
+		expect(component.timeBoundValue).toBe('Past week');
+	}));
+
+	it('should call getTimeBoundResults(else)', async(() => {
+		const fixture = TestBed.createComponent(FeedAdvancedSearchComponent);
+		const component = fixture.debugElement.componentInstance;
+		component.getTimeBoundResults('else');
+
+		expect(component.timeBoundValue).toBe('Any time');
+	}));
+
+	it('should call getLocationBasedResults(all)', async(() => {
+		const fixture = TestBed.createComponent(FeedAdvancedSearchComponent);
+		const component = fixture.debugElement.componentInstance;
+		component.getLocationBasedResults('all');
+
+		expect(component.locationValue).toBe('All Countries');
+	}));
+
+	it('should call getLocationBasedResults(India)', async(() => {
+		const fixture = TestBed.createComponent(FeedAdvancedSearchComponent);
+		const component = fixture.debugElement.componentInstance;
+		component.getLocationBasedResults('India');
+
+		expect(component.locationValue).toBe('Country: India');
+	}));
+
+	it('should call getLocationBasedResults(China)', async(() => {
+		const fixture = TestBed.createComponent(FeedAdvancedSearchComponent);
+		const component = fixture.debugElement.componentInstance;
+		component.getLocationBasedResults('China');
+
+		expect(component.locationValue).toBe('Country: China');
+	}));
+
+	it('should call getLocationBasedResults(US)', async(() => {
+		const fixture = TestBed.createComponent(FeedAdvancedSearchComponent);
+		const component = fixture.debugElement.componentInstance;
+		component.getLocationBasedResults('US');
+
+		expect(component.locationValue).toBe('Country: US');
+	}));
+
+	it('should call getLocationBasedResults(UK)', async(() => {
+		const fixture = TestBed.createComponent(FeedAdvancedSearchComponent);
+		const component = fixture.debugElement.componentInstance;
+		component.getLocationBasedResults('UK');
+
+		expect(component.locationValue).toBe('Country: UK');
+	}));
+
+	it('should call getLocationBasedResults(else)', async(() => {
+		const fixture = TestBed.createComponent(FeedAdvancedSearchComponent);
+		const component = fixture.debugElement.componentInstance;
+		component.getLocationBasedResults('else');
```

Comment:
Can you explain the logic behind `component.getLocationBasedResults('else'); ` ? What is the scenario that you are testing?